apps/meteor/client/views/room/components/body/composer/ComposerReadOnly.tsx

```diff
@@ -1,9 +1,42 @@
-import { MessageFooterCallout } from '@rocket.chat/ui-composer';
+import { MessageFooterCallout, MessageFooterCalloutContent } from '@rocket.chat/ui-composer';
 import { useTranslation } from '@rocket.chat/ui-contexts';
 import type { ReactElement } from 'react';
-import React from 'react';
+import React, { useCallback, useRef, useEffect } from 'react';
+
+import { call } from '../../../../../lib/utils/call';
+import { useRoom, useUserIsSubscribed } from '../../../contexts/RoomContext';
 
 export const ComposerReadOnly = (): ReactElement => {
 	const t = useTranslation();
-	return <MessageFooterCallout>{t('room_is_read_only')}</MessageFooterCallout>;
+	const room = useRoom();
+	const isSubscribed = useUserIsSubscribed();
+	const calloutRef = useRef() as React.MutableRefObject<HTMLButtonElement>;
+
+	useEffect(() => {
+		calloutRef.current.style.flex = !isSubscribed ? '4' : 'none';
+	}, [isSubscribed]);
+
+	const join = useCallback(
+		async (_e) => {
+			try {
+				await call('joinRoom', room._id);
```

Comment:
Can you try to use the REST API instead here instead of the meteor call? We're trying to move away from using meteor and rather use standard HTTP practices

Reply:
I can see there's this endpoint that u can use instead -> `channels.join`

Reply:
I firstly used the same endpoint, then switched to meteor call, have fixed it now.